code/controllers/subsystem/puddle.dm

```diff
@@ -0,0 +1,23 @@
+SUBSYSTEM_DEF(puddle)
+	name = "Puddle"
+	wait = 5
+	runlevels = RUNLEVEL_GAME
+	var/list/puddlelist
+
+/datum/controller/subsystem/puddle/Initialize(start_timeofday)
+	. = ..()
+	puddlelist = list()
+
+/datum/controller/subsystem/puddle/stat_entry(msg)
+	..("P: [length(puddlelist)]")
+
+/datum/controller/subsystem/puddle/fire(resumed)
+	// used on the machines subsystem, apparently makes this faster
+	var/list/pudl = src.puddlelist
+	
+	for(var/obj/puddle/puddle in pudl)
```

Comment:
```suggestion
	for(var/obj/puddle/puddle in puddlelist)
```
This also ought to be a processing subsystem subtype, as those are built for handling process() calls on objects better, no?